eox_nelp/stats/tests/api/v1/tests_views.py

```diff
@@ -0,0 +1,230 @@
+
+        response = self.client.get(url_endpoint)
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertTrue(["learners", "courses", "instructors", "components"] == list(response.data.keys()))
+
+    @override_settings(
+        MIDDLEWARE=["eox_tenant.middleware.CurrentSiteMiddleware"],
+        STATS_SETTINGS={"API_XBLOCK_TYPES": ["html", "problem", "video"]},
+    )
+    @patch("eox_nelp.stats.api.v1.views.metrics")
+    def test_total_components(self, mock_metrics):
+        """
+        Test that the view will calculate the total of components based on the metrics values
+
+        Expected behavior:
+            - Status code 200.
+            - Components total values are the expected.
+        """
+        fake_metric = {
+            "components": {
+                "html": 5,
+                "problem": 10,
+                "video": 15,
+            }
+        }
+        mock_metrics.get_learners_metric.return_value = 5
+        mock_metrics.get_instructors_metric.return_value = 4875
+        mock_metrics.get_courses_metrics.return_value = {
+            "total_courses": 3,
+            "metrics": [
+                fake_metric, fake_metric, fake_metric, fake_metric
+            ],
+        }
+        url_endpoint = reverse("stats-api:v1:general-stats")
+
+        response = self.client.get(url_endpoint)
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual({"html": 20, "problem": 40, "video": 60}, response.data["components"])
+
+    @override_settings(MIDDLEWARE=["eox_tenant.middleware.CurrentSiteMiddleware"])
+    @data("post", "put", "patch", "delete")
+    def test_invalid_method(self, method):
+        """
+        This test that the view returns a method not allowed response, since the get is the unique valid method.
+
+        Expected behavior:
+            - Status code 405.
+        """
+        url_endpoint = reverse("stats-api:v1:general-stats")
+        request = getattr(self.client, method)
+
+        response = request(url_endpoint)
+
+        self.assertEqual(status.HTTP_405_METHOD_NOT_ALLOWED, response.status_code)
+
+
+@ddt
+class GeneralCourseStatsViewTestCase(APITestCase):
+    """ Test GeneralCourseStatsView."""
+
+    def setUp(self):
+        """
+        Create site since the view use the request.site attribute to determine the current domain.
+        """
+        Site.objects.get_or_create(domain="testserver")
+
+    @override_settings(MIDDLEWARE=["eox_tenant.middleware.CurrentSiteMiddleware"])
+    @patch("eox_nelp.stats.api.v1.views.metrics")
+    def test_get_list(self, mock_metrics):
+        """
+        Test a get request, this will verify the standard view behavior by checking the call of the metrics functions.
+
+        Expected behavior:
+            - Status code 200.
+            - the total_course key is present and has the expected value
+            - the lenght of metrics is the expected.
```

Comment:
```suggestion
            - the length of metrics expected.
```

Reply:
indeed that should be `the length of metrics is as expected`